assembly/src/parsers/blocks.rs

```diff
@@ -251,22 +254,45 @@ impl BlockParser {
 // ================================================================================================
 
 pub fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
-    // merge consecutive Span blocks
+    // merge consecutive Span blocks.
     let mut merged_blocks: Vec<CodeBlock> = Vec::with_capacity(blocks.len());
+    // Keep track of all the consecutive Span blocks and are merged together when there is a discontinuity.
+    let mut continuous_span_vector: Vec<Span> = Vec::new();
+    // flag for last block being a SPAN block or not.
+    let mut is_last_block_span = false;
+
     blocks.drain(0..).for_each(|block| {
         if block.is_span() {
-            if let Some(CodeBlock::Span(last_span)) = merged_blocks.last_mut() {
-                // this is guaranteed to execute because we know that the block is a span
-                if let CodeBlock::Span(span) = block {
-                    last_span.append(span);
-                }
-            } else {
+            // This is to avoid redoing the same exact compution already done earlier while creating a
+            // SPAN block of operational batches from exported procedure body.
+            if merged_blocks.capacity() == 1 {
                 merged_blocks.push(block);
             }
```

Comment:
I found this reliance on checking capacity a bit difficult to understand, and I think it doesn't handle situations where we have something like `<non-span> <span> <non-span>`. Basically, the `SPAN` in the middle would still be re-built from individual ops.

Maybe the following approach would be better:
* Make `continuous_span_vector` store `CodeBlock`'s rather than `Span`'s. We will actually put only `SPAN` blocks into it, but it should prevent you from needing to get inner values of `CodeBlock`'s.
* Always add a span block to the `continuous_span_vector`.
* Have a function - something like `combine_spans()` which would take a vector of `CodeBlock`'s (all of which must be `Span`'s, and returns a single `CodeBlock` which would be a `Span`.

Reply:
I've introduced a new function in `CodeBlock` module which accepts a span and returns a `CodeBlock` with that span. That way it never has to do any re-computation on the operations of the provide span block.